File: goormlevel/174784/1. 밀도 정렬/밀도 정렬.py
n = int(input())

answer = 0
maxval = 0
weight = 0
for i in range(1, n + 1):
	w, v = map(int, input().split())

	m = w / v

	if maxval < m:
		maxval = m
		answer = i
		weight = w
	elif maxval == m:
		if weight < w:
			weight = w
			answer = i

print(answer)

File: goormlevel/174784/1. 밀도 정렬/밀도 정렬.cpp
#include <iostream>
using namespace std;
int main() {
	int n = 0;
	if (!(cin >> n)) return 0;

	int answer = 0;
	double maxval = -1.0;
	int weight = 0;
	for (int i = 1; i <= n; i++) {
		int w = 0, v = 0;
		if (!(cin >> w >> v)) return 0;

		double m = (double)w / v;

		if (maxval < m) {
			maxval = m;
			answer = i;
			weight = w;
		}
		else if (maxval == m) {
			if (weight < w) {
				weight = w;
				answer = i;
			}
		}
	}

	cout << answer << "\n";

	return 0;
}